### PR TITLE
Change formatting of lecture notes

### PR DESCRIPTION
**Pull request by dahalankur**
Corrected typos and fixed formatting issue from lecture notes.

### CODE REVIEW

File: lecture-notes/1-cli.md [hunk 1]
```diff
@@ -871,12 +871,12 @@ See also the `screen` command, which is similar.
 
 #### `vi`
 `vi` is a POSIX-specified text editor that is available on almost every system
-you will use. Unlike text editors such as Notepad and Kate, is a *modal*
+you will use. Unlike text editors such as Notepad and Kate, `vi` is a *modal*
```
Comment (tchebb) [inline lecture-notes/1-cli.md:874]: Can you style this as "Vi" instead of `vi`, like we do for other mentions of the name later?

**Review by tchebb — CHANGES_REQUESTED** [review]
Good catches, thanks for the contribution! One more nit:

> Made formatting changes to lecture notes

In this repository, we prefer commit messages written with an imperative style, for example "Make formatting changes to lecture notes", or—more concisely—"Change formatting of lecture notes".

This is a common convention for commit messages and follows the example set by the Linux Kernel, which Git was written for. Its [Submitting Patches](https://www.kernel.org/doc/html/latest/process/submitting-patches.html#describe-your-changes) guide says the following:
> Describe your changes in imperative mood, e.g. “make xyzzy do frotz” instead of “[This patch] makes xyzzy do frotz” or “[I] changed xyzzy to do frotz”, as if you are giving orders to the codebase to change its behaviour.

File: lecture-notes/1-cli.md (hunk 1 above)
Comment (dahalankur) [reply lecture-notes/1-cli.md]: Done!

**Comment by tekknolagi** [comment]
Note that changing the PR title does not change your commit title

**Comment by tchebb** [comment]
> Thanks for the contribution! Note that changing the PR title does not change your commit title

To elaborate on this, a Pull Request on GitHub represents a branch you want us to merge, and that branch can contain as many commits as you want. In this case, instead of editing your original commit, you added a new commit to address my comment about "Vi" styling, as you can see in the "Commits" tab.

This isn't inherently a bad thing to do; your workflow of adding a new commit for every revision of the pull request is in fact what GitHub's UI encourages and what its documentation [suggests](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#about-pull-requests):

> Once you've created a pull request, you can push commits from your topic branch to add them to your existing pull request. These commits will appear in chronological order within your pull request and the changes will be visible in the "Files changed" tab.

There are benefits of adding a new commit for each revision you make to a PR:
 - It's easy to see how it's been revised over time
 - Reviewers can leave comments on lines that change from one revision to the next
 - Each commit can hold notes about why the revision was necessary

However, these exact same things become drawbacks once the pull request is actually merged:
 - Mistakes caught during code review are nonetheless immortalized in the repository's history, even if those mistakes cause confusion to readers or break the build (which affects more advanced Git features like `git bisect`)
 - Bad commit messages can't be fixed—they can only be clarified in a follow-up commit, which might never be seen by someone finding the commit through `git blame`
 - The final set of changes in the PR, which probably represents just a few conceptual changes to the code, ends up fragmented across many commits

So, what's the alternative to this workflow if you (like I) think the drawbacks outweigh the benefits? One solution is to use GitHub's "[Squash and merge](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits)" feature. When a PR is merged in this mode, GitHub's backend generates a single new commit containing all the changes from the PR's branch whose title and description match those of the PR's. For this PR specifically, this mode would work great and your PR title change would be all that's needed to address my feedback.

Unfortunately, we have that mode disabled for this repo (although I might ask Max to turn it back on). The reason we disabled it is because I feel like it's too limiting for more complex PRs that contain multiple independent changes. For just a couple examples of this from my own PR history, see sigrokproject/libsigrok#145 and kaitai-io/kaitai_struct_formats#514. Each of these PRs contains a related set of commits, and it would be burdensome and spammy to make each one a separate PR; however, it would be just as bad to merge either PR as a single commit, since the commits as I wrote them each represent a separate logical change and have detailed messages justifying that change.

This philosophy of "each commit should represent one logical change in its final form" isn't unique to me. To again quote the Linux kernel [patch guidelines](https://www.kernel.org/doc/html/latest/process/submitting-patches.html#separate-your-changes):
> Separate each logical change into a separate patch.
>
> For example, if your changes include both bug fixes and performance enhancements for a single driver, separate those changes into two or more patches. If your changes include an API update, and a new driver which uses that new API, separate those into two patches.
>
> […]
>
> The point to remember is that each patch should make an easily understood change that can be verified by reviewers. Each patch should be justifiable on its own merits.

While this workflow isn't encouraged by GitHub's UI, it's still possible by using *force pushes*. My personal workflow for revising my GitHub PRs is to use `git amend` and `git rebase -i` to incorporate review feedback into the appropriate existing commits on my branch, then repush that edited history to GitHub with `git push -f`.

For the moment, that's also what you'll have to do to fully address my feedback. However, I can't fault you for doing what GitHub recommends, even if that's at odds to how many people use Git outside of GitHub. So I'm also happy to have Max turn "Squash and merge" back on and then merge this PR that way. :)

**Comment by dahalankur** [comment]
> While this workflow isn't encouraged by GitHub's UI, it's still possible with _force pushes_. My personal workflow for revising my GitHub PRs is to use `git amend` and `git rebase -i` to incorporate review feedback into the appropriate existing commits on my branch and then push that edited history to GitHub with `git push -f`.
> 
> For the moment, that's also what you'll have to do to fully address my feedback. However, I can't fault you for doing what GitHub recommends, even if that's at odds to how many people use Git outside of GitHub. So I'm also happy to have Max turn "Squash and merge" back on and merge this PR that way. :)

Thank you for the feedback! This was an amazing learning experience. I have made changes according to your suggestions and pushed a single commit replacing my previous commits. Please let me know if this is better.

**Comment by tchebb** [comment]
Looks great! Merged.

**Pull request merged**